net/dns/resolver/forwarder.go

```diff
@@ -303,12 +313,34 @@ func resolversWithDelays(resolvers []*dnstype.Resolver) []resolverAndDelay {
 // The memory referenced by routesBySuffix should not be modified.
 func (f *forwarder) setRoutes(routesBySuffix map[dnsname.FQDN][]*dnstype.Resolver) {
 	routes := make([]route, 0, len(routesBySuffix))
+
+	var cloudResolver []*dnstype.Resolver
+	var cloudHostFallback []resolverAndDelay
+	if ip := cloudenv.Get().ResolverIP(); ip != "" {
+		cloudResolver = []*dnstype.Resolver{{Addr: ip}}
+		cloudHostFallback = resolversWithDelays(cloudResolver)
```

Comment:
Do sync.Once and keep it around? The cloud env isn't going to change over time, so it can be done once per process.

Reply:
done